example/multiple_drones/multiple_drones.cpp

```diff
@@ -0,0 +1,153 @@
+//
+// Example to connect multiple vehicles and make them take off and land in parallel.
+//./multiple_drones udp://:14540 udp://:14541
+//
+// Author: Julian Oes <julian@oes.ch>
+// Author: Shayaan Haider (via Slack)
+
+#include <dronecode_sdk/dronecode_sdk.h>
+#include <dronecode_sdk/plugins/action/action.h>
+#include <dronecode_sdk/plugins/telemetry/telemetry.h>
+#include <chrono>
+#include <cstdint>
+#include <iostream>
+#include <thread>
+
+using namespace dronecode_sdk;
+using namespace std::this_thread;
+using namespace std::chrono;
+
+static void takeoff_land(System &system);
+
+#define ERROR_CONSOLE_TEXT "\033[31m" // Turn text on console red
+#define TELEMETRY_CONSOLE_TEXT "\033[34m" // Turn text on console blue
+#define NORMAL_CONSOLE_TEXT "\033[0m" // Restore normal console colour
+
+int main(int argc,
+         char **argv) // The input looks like this ./multiple_drones udp://:14540 udp://:14541
+{
+    if (argc == 1) {
+        std::cout << ERROR_CONSOLE_TEXT << "Please specify connection" << NORMAL_CONSOLE_TEXT
+                  << std::endl;
+        ;
+        return 1;
+    }
+
+    DronecodeSDK dc;
+
+    // the loop below adds the number of ports the sdk monitors.
+    for (int i = 1; i < argc; ++i) {
+        ConnectionResult connection_result = dc.add_any_connection(argv[i]);
+        if (connection_result != ConnectionResult::SUCCESS) {
+            std::cout << ERROR_CONSOLE_TEXT
+                      << "Connection error: " << connection_result_str(connection_result)
+                      << NORMAL_CONSOLE_TEXT << std::endl;
+            ;
+            return 1;
```

Comment:
So if any of the port is already busy, the example exits ? Lets say I am controlling 3 drones, and 1 of them is QGC port! So, can't I just continue with those two of them ?

Reply:
Good point, thanks. That can be improved!

Reply:
I think I'd leave it like that because it's really something unexpected and it will make it harder to debug why things are not working after that.